### PR TITLE
Update tests (issue #76)

### PR DESCRIPTION
**Pull request by yolaj-nhs**
Improve the tests by doing the following:

- [x] Test the contents of the attribute inference attack in test_attribute_inference_attack.py
- [x] In test_generate_report.py - test for the contents of the generated file (not just whether they pass/fail)
- [x] In test_metrics.py - check the results of the calculation, not just pass/fail
- [x] Remove test_safekeras.old
- [x] The GenerateJSON module is tested inside other tests, which would make things very unclear if there was a failure. Refactor this so the tests are happening in the right place
- [x] test_lira_target should be cleaned after tests
- [x] Removing aisdc/safemodel/classifiers/dp_svc.py warning
- [x] Cleaning files that have been created in tests

Any suggestions of other improvements that could be made to the tests are welcome, and can also be added to the following ticket: 

https://github.com/AI-SDC/AI-SDC/issues/76

### CODE REVIEW

**Comment by yolaj-nhs** [comment]
> 

Done!

File: tests/test_attribute_inference_attack.py [hunk 1]
```diff
@@ -31,110 +33,140 @@
...
+            "aia_example.json",
+            "aia_example.pdf",
+            "aia_report_cat_frac.png",
+            "aia_report_cat_risk.png",
+            "aia_report_quant_risk.png",
+            "aia_report.pdf",
+            "aia_report.json",
+            "aia_attack_from_configfile.json",
+            "test_attribute_attack.json",
+            "tests/test_config_aia_cmd.json",
```
Comment (simonrnss) [inline tests/test_attribute_inference_attack.py:167]: hard-coded dir seps

Comment (simonrnss) [reply tests/test_attribute_inference_attack.py:167]: and below

**Review by rpreen — DISMISSED** [review]
This PR has a few merge conflicts now that need to be addressed because of changes in other PRs.

File: tests/test_lira_attack.py [hunk 2]
```diff
@@ -254,12 +254,14 @@ def tearDownClass(cls):
         names = [
             "test_output_lira/",
```
Comment (simonrnss) [inline tests/test_lira_attack.py:255]: do we need the slashes? And if so, `os.sep`

File: tests/test_attack_report_formatter.py [hunk 3]
```diff
@@ -257,12 +264,8 @@ def test_move_files(self):
 
         clean_up(filename)
         clean_up(output_filename)
-        clean_up(os.path.join("release_dir", dummy_model))
-        clean_up(os.path.join("release_dir", filename))
-        clean_up(os.path.join("release_dir", output_filename))
-        clean_up("release_dir")
-        clean_up(os.path.join("training_artefacts", png_file))
-        clean_up("training_artefacts")
+        clean_up("release_dir/")
```
Comment (simonrnss) [inline tests/test_attack_report_formatter.py:267]: slash the slash

**Comment by simonrnss** [comment]
Sorry - vague comments...I'd prefer not to hard-code `/`, but I'm not sure if the `is.dir` will work without it? So, feel free to leave them if it's an issue.

**Comment by yolaj-nhs** [comment]
> This PR has a few merge conflicts now that need to be addressed because of changes in other PRs.

Merge conflicts have been resolved

**Pull request merged**